src/list/list.spec.ts

```diff
@@ -0,0 +1,224 @@
+    expect(list.find((content) => content === 'panqueque')).toEqual('panqueque');
+  });
+
+  it('should return undefined after not finding the element in the list', () => {
+    const elementsMock = ['Calabaza', 'Pan', 'Calabaza', 'Salchicha', 'panqueque'];
+    elementsMock.forEach((element) => list.add(element));
+    expect(list.find((content) => content === 'sandwitch')).toEqual([]);
+  });
+
+  it('should return an array with the elements found in the list', () => {
+    const elementsMock = ['Calabaza', 'Pan', 'Calabaza', 'Salchicha', 'panqueque'];
+    elementsMock.forEach((element) => list.add(element));
+    expect(list.find((content) => content === 'Calabaza', true)).toEqual(['Calabaza', 'Calabaza']);
+  });
+
+  it('should return undefined after not finding items in the list', () => {
+    const elementsMock = ['Calabaza', 'Pan', 'Calabaza', 'Salchicha', 'panqueque'];
+    elementsMock.forEach((element) => list.add(element));
+    expect(list.find((content) => content === 'sandwich', true)).toEqual([]);
+  });
+
+  it('should iterate through all element in the list', () => {
+    const elementsMock = ['paquito', 'alfajor'];
+    const addedPoint: String[] = [];
+    function addPoint(element: any): void {
+      addedPoint.push(`${element}.`);
+    }
+    elementsMock.forEach((element) => list.add(element));
+    list.forEach(addPoint);
+    for (let i = 0; i < elementsMock.length - 1; i += 1) {
+      expect(addedPoint[i]).toEqual(`${elementsMock[i]}.`);
+    }
+  });
+
+  it('should return a new empty list', () => {
+    const addedPoint: String[] = [];
+    function addPoint(element: any): any {
+      addedPoint.push(`${element}.`);
+      return (`${element}.`);
+    }
+    const result = list.map(addPoint);
+    expect(result.length).toEqual(0);
+  });
+
+  it('should return a new list with all mapped elements', () => {
+    const elementsMock = ['paquito', 'alfajor'];
+    const addedPoint: String[] = [];
+    function addPoint(element: any): any {
+      addedPoint.push(`${element}.`);
+      return (`${element}.`);
+    }
+    elementsMock.forEach((element) => list.add(element));
+    const result = list.map(addPoint);
+    expect(result).toBeInstanceOf(List);
+    for (let i = 0; i < elementsMock.length - 1; i += 1) {
+      expect(addedPoint[i]).toEqual(`${elementsMock[i]}.`);
+    }
+  });
+
+  it('should filter and return an empty list', () => {
+    const filteredList: List<String> = list.filter((content) => content === 'alfajor');
+    expect(filteredList.length).toEqual(0);
+  });
+
+  it('should filter and return a list with two elements', () => {
+    const elementsMock = ['paquito', 'alfajor', 'sandia', 'peras', 'alfajor'];
+    elementsMock.forEach((element) => list.add(element));
+    const filteredList: List<String> = list.filter((content) => content === 'alfajor');
+    expect(filteredList.length).toEqual(2);
+  });
+
+  it('should reduce the list and return an string', () => {
```

Comment:
create a reduce test when no initial value is provided (it should use the first element in the list) 

See the description from the docs of array:

A value to which previousValue is initialized the first time the callback is called. If initialValue is specified, that also causes currentValue to be initialized to the first value in the array. If initialValue is not specified, previousValue is initialized to the first value in the array, and currentValue is initialized to the second value in the array.